### PR TITLE
Bug: deserialized scene should have node transforms invalidated at initial

### PR DESCRIPTION
**Pull request by shrinktofit**
Re: #16352

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by shrinktofit** [comment]
@cocos-robot run test cases

File: native/cocos/core/scene-graph/Node.h [hunk 1]
```diff
@@ -657,7 +657,7 @@ class Node : public CCObject {
     // NOTE: TypeArray created in node.jsb.ts _ctor should have the same memory layout
     uint32_t _eventMask{0};                                             // Uint32: 0
     uint32_t _layer{static_cast<uint32_t>(Layers::LayerList::DEFAULT)}; // Uint32: 1
-    uint32_t _transformFlags{0};                                        // Uint32: 2
+    uint32_t _transformFlags{static_cast<uint32_t>(TransformBit::TRS)}; // Uint32: 2
```
Comment (dumganhar) [inline native/cocos/core/scene-graph/Node.h:660]: ` this._transformFlags |= TransformBit.TRS;` was set in `_onBatchCreated`.

If we modify the initialized value to TRS, will it mark it as dirty twice and call `updateWorldTransform` twice?

For instance,
```ts
const scene = deserialize(archive) as Scene;
const node = scene.getChildByPath('Node')!;
const worldScale = node.worldScale;  // call updateWorldTransform internally, and reset child._transformFlags = TransformBit.NONE;

// Use worldScale 

director.runScene(scene); // call scene._load -> scene._onBatchCreated -> node._onBatchCreated -> this._transformFlags |= TransformBit.TRS; 
//  this._transformFlags is set to TransformBit.TRS again, so updateWorldTransform will be invoked twice, but in fact, the world transform was not changed at all.

```

Comment (shrinktofit) [reply native/cocos/core/scene-graph/Node.h:660]: I'd recommend deleting transform flags modification in `Node._onBatchCreated`.

Comment (dumganhar) [reply native/cocos/core/scene-graph/Node.h:660]: That's ok for me.

**Review by dogeFu — APPROVED** [review]

File: native/cocos/core/scene-graph/Node.h (hunk 1 above)
Comment (shrinktofit) [reply native/cocos/core/scene-graph/Node.h:660]: https://github.com/cocos/cocos-engine/pull/16353/commits/abb7ea2d65960b25ca3f3e0a5a72bc5514874fd3

**Comment by shrinktofit** [comment]
@cocos-robot run test cases

**Review by SantyWang — APPROVED** [review]

**Review by dumganhar — APPROVED** [review]

**Pull request merged**